packages/wonder-blocks-data/src/util/__tests__/response-cache.test.js

```diff
@@ -436,105 +459,371 @@ describe("../response-cache.js", () => {
-            jest.spyOn(internalCache, "retrieve").mockReturnValue({
-                data: "data!",
+        describe("when server-side", () => {
+            beforeEach(() => {
+                jest.spyOn(Server, "isServerSide").mockReturnValue(true);
             });
-            const cache = new ResponseCache(internalCache);
-            const fakeHandler: IRequestHandler<string, string> = {
-                getKey: () => "MY_KEY",
-                type: "MY_HANDLER",
-                fulfillRequest: jest.fn(),
-                hydrate: true,
-            };
 
-            // Act
-            const result = cache.getEntry(fakeHandler, "options");
+            describe("handler wants hydration", () => {
+                it("should return null if not in the hydration cache", () => {
+                    // Arrange
+                    const hydrationCache = new MemoryCache();
+                    jest.spyOn(hydrationCache, "retrieve").mockReturnValue(
+                        null,
+                    );
+                    const cache = new ResponseCache(hydrationCache);
+                    const fakeHandler: IRequestHandler<string, string> = {
+                        getKey: () => "MY_KEY",
+                        type: "MY_HANDLER",
+                        fulfillRequest: jest.fn(),
+                        hydrate: true,
+                    };
 
-            // Assert
-            expect(result).toStrictEqual({data: "data!"});
+                    // Act
+                    const result = cache.getEntry(fakeHandler, "options");
+
+                    // Assert
+                    expect(result).toBeNull();
+                });
+
+                it("should return the cached entry if in the hydration cache", () => {
+                    // Arrange
+                    const hydrationCache = new MemoryCache();
+                    jest.spyOn(hydrationCache, "retrieve").mockReturnValue({
+                        data: "data!",
+                    });
+                    const cache = new ResponseCache(hydrationCache);
+                    const fakeHandler: IRequestHandler<string, string> = {
+                        getKey: () => "MY_KEY",
+                        type: "MY_HANDLER",
+                        fulfillRequest: jest.fn(),
+                        hydrate: true,
+                    };
+
+                    // Act
+                    const result = cache.getEntry(fakeHandler, "options");
+
+                    // Assert
+                    expect(result).toStrictEqual({data: "data!"});
+                });
+            });
+
+            describe("handler does not want hyrdation", () => {
+                it("should return null if not in the ssr-only cache", () => {
+                    // Arrange
+                    const hydrationCache = new MemoryCache();
+                    const ssrOnlyCache = new MemoryCache();
+                    jest.spyOn(ssrOnlyCache, "retrieve").mockReturnValue(null);
```

Comment:
Do we not bother looking in the `hydrationCache` if the `ssrOnlyCache` returns `null`?

Reply:
We don't even look in the hydrationCache if the handler says it doesn't want hydration, which is what these tests are all about.

Reply:
I've made a change here to mock out hydrationCache (even though it shouldn't be referenced) so that we catch if the `hydrate` value below is wrong, since it was wrong, but the test had passed. Now if it's wrong, the test will fail.